### PR TITLE
OCPQE-26834 [WIP] create capi GCP machineset 

### PR DESCRIPTION
**Pull request by miyadav**
@sunzhaohua2 @huali9 I am following the pattern used in earlier PR for Azure (zhaohua) , not able to connect with the cluster yet , will explore further , if time permits , have a look and suggest what might be missing , `make all ` is good ..

I exported KUBECONFIG

but still test unable to execute 
```
`miyadav@miyadav-thinkpadx1carbongen8:~/cluster-api-actuator-pkg$ ./hack/ci-integration.sh -focus "Cluster API GCP MachineSet" -v  
Running Suite: Machine Suite - /home/miyadav/cluster-api-actuator-pkg/pkg
=========================================================================
Random Seed: 1730963288

Will run 0 of 39 specs
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
[ReportAfterSuite] Autogenerated ReportAfterSuite for --junit-report
autogenerated by Ginkgo
[ReportAfterSuite] PASSED [0.003 seconds]
------------------------------

Ran 0 of 39 Specs in 0.001 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 39 Skipped
PASS

Ginkgo ran 1 suite in 6.062309947s
Test Suite Passed
`
```

### CODE REVIEW

**Comment by huali9** [comment]
@miyadav I think you miss this line https://github.com/openshift/cluster-api-actuator-pkg/pull/337/commits/c96266b8930dd627ea1ec3943dce4dc36586536e#diff-2752ea6ec73540b0edb4d7fb0eb3996fc342d662ddd5a4fc4c76b7671dab244dR21

**Comment by miyadav** [comment]
> @miyadav I think you miss this line [c96266b#diff-2752ea6ec73540b0edb4d7fb0eb3996fc342d662ddd5a4fc4c76b7671dab244dR21](https://github.com/openshift/cluster-api-actuator-pkg/commit/c96266b8930dd627ea1ec3943dce4dc36586536e#diff-2752ea6ec73540b0edb4d7fb0eb3996fc342d662ddd5a4fc4c76b7671dab244dR21)

made the commit still same :(

File: pkg/capi/gcp.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
+package e2e
```
Comment (huali9) [inline pkg/capi/gcp.go:1]: You should update the package from `e2e` to `capi`

**Comment by miyadav** [comment]
@sunzhaohua2 @huali9 @damdo @RadekManak PTAL when time permits , adding multiple disks test ( the last commit )  , will put test case numbers as well , to track them in QE reports.

```
`miyadav@miyadav-thinkpadx1carbongen8:~/cluster-api-actuator-pkg$ ./hack/ci-integration.sh -focus "Cluster API GCP MachineSet" -v  
Running Suite: Machine Suite - /home/miyadav/cluster-api-actuator-pkg/pkg
=========================================================================
Random Seed: 1731677321

Will run 3 of 44 specs
------------------------------
[BeforeSuite] 
/home/miyadav/cluster-api-actuator-pkg/pkg/e2e_test.go:56
[BeforeSuite] PASSED [3.474 seconds]
------------------------------
SSSSSSSSSSSSSSS
------------------------------
Cluster API GCP MachineSet should be able to run a machine with disk type: pd-standard
/home/miyadav/cluster-api-actuator-pkg/pkg/capi/gcp.go:82
  STEP: Creating core cluster @ 11/15/24 18:58:57.117
  STEP: Creating GCP machine template @ 11/15/24 18:59:01.64
  STEP: Creating MachineSet "gcp-machineset" @ 11/15/24 18:59:06.664
  STEP: Waiting for MachineSet machines "gcp-machineset" to enter Running phase @ 11/15/24 18:59:07.11
  STEP: Deleting MachineSet "gcp-machineset" @ 11/15/24 19:04:00.008
  STEP: Waiting for MachineSet "gcp-machineset" to be deleted @ 11/15/24 19:04:01.063
  STEP: Deleting /gcp-machine-template @ 11/15/24 19:04:42.41
• [347.506 seconds]
------------------------------
Cluster API GCP MachineSet should be able to run a machine with disk type: pd-ssd
/home/miyadav/cluster-api-actuator-pkg/pkg/capi/gcp.go:82
  STEP: Creating GCP machine template @ 11/15/24 19:04:43.753
  STEP: Creating MachineSet "gcp-machineset" @ 11/15/24 19:04:44.166
  STEP: Waiting for MachineSet machines "gcp-machineset" to enter Running phase @ 11/15/24 19:04:44.451
  STEP: Deleting MachineSet "gcp-machineset" @ 11/15/24 19:08:00.87
  STEP: Waiting for MachineSet "gcp-machineset" to be deleted @ 11/15/24 19:08:01.151
  STEP: Deleting /gcp-machine-template @ 11/15/24 19:08:39.166
• [237.027 seconds]
------------------------------
Cluster API GCP MachineSet should be able to run a machine with disk type: pd-balanced
/home/miyadav/cluster-api-actuator-pkg/pkg/capi/gcp.go:82
  STEP: Creating GCP machine template @ 11/15/24 19:08:40.49
  STEP: Creating MachineSet "gcp-machineset" @ 11/15/24 19:08:41.081
  STEP: Waiting for MachineSet machines "gcp-machineset" to enter Running phase @ 11/15/24 19:08:41.667
  STEP: Deleting MachineSet "gcp-machineset" @ 11/15/24 19:14:33.832
  STEP: Waiting for MachineSet "gcp-machineset" to be deleted @ 11/15/24 19:14:34.343
  STEP: Deleting /gcp-machine-template @ 11/15/24 19:15:29.191
• [409.763 seconds]
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
[ReportAfterSuite] Autogenerated ReportAfterSuite for --junit-report
autogenerated by Ginkgo
[ReportAfterSuite] PASSED [0.004 seconds]
------------------------------

Ran 3 of 44 Specs in 997.772 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 41 Skipped
PASS

Ginkgo ran 1 suite in 16m47.696208408s
Test Suite Passed`
```

File: pkg/capi/gcp.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		}
+		framework.DeleteCAPIMachineSets(ctx, cl, machineSet)
+		framework.WaitForCAPIMachineSetsDeleted(ctx, cl, machineSet)
+		framework.DeleteObjects(ctx, cl, gcpMachineTemplate)
+	})
+
+	diskTypes := map[string]gcpv1.DiskType{
+		"pd-standard": gcpv1.PdStandardDiskType,
+		"pd-ssd":      gcpv1.PdSsdDiskType,
+		//	"local-ssd":   gcpv1.LocalSsdDiskType,//boot disk cannot be local disk hence commenting it
```
Comment (miyadav) [inline pkg/capi/gcp.go:77]: @damdo @RadekManak , do we want to have a webhook like in `mapi`

`error: machinesets.machine.openshift.io "miyadav-1811g-vk7vb-worker-f" could not be patched: admission webhook "validation.machineset.machine.openshift.io" denied the request: providerSpec.disks[0].type: Unsupported value: "local-ssd": supported values: "hyperdisk-balanced", "pd-balanced", "pd-ssd", "pd-standard"
`

**Comment by miyadav** [comment]
Added ShieldedInstanceConfigs

**Pull request closed without merging**